rfcs/0007-synchronous-commit.md

```diff
@@ -204,68 +202,43 @@ Let's assume we have a sequence of writes like this:
 
-- The "last committed watermark"'s position (in term of sequence number) is always greater than or equal to the "last local persisted watermark"
-- The "last local persisted watermark"'s position is always greater than or equal to the "last remote persisted watermark"
+By default, no matter what `DurabilityLevel` is set in the `durability_filter` option, the read operation will always read the Committed data. But at some use cases like disk caching, users may not care about the commitness of the data, but only care about the data is durable at least at `DurabilityLevel::Local`.
 
-If we successfully write a key that succesfully persisted to S3, at this point, all these watermarks will point to the same position. 
+If the writer write with `SyncLevel::Remote`, the data will not be considered as committed until it's persisted to `Remote`. If the user want to read the data as soon as it's persisted to `Local`, they can use an additional option `dirty: true` to make this possible:
 
-For reads in a transaction, it's possible to allow users specify the watermark as `ReadWatermark::LastRemotePersisted` or `ReadWatermark::LastLocalPersisted`. This might let users read an older version of the data, but it won't violate the semantics of conflict checking: if there exists a newer version of the same key, this transaction could be rolled back on commit.
-
-In conclusion, the difference between the current model and this proposal is adding a "Committed" watermark. And as it does not make sense to use `DurabilityLevel` to contain it, we propose to use `ReadWatermark` to identify the positions of the commit history for read operations.
-
-In most of the use cases, users do not need to care about this `ReadWatermark` option. `ReadWatermark::LastCommitted` is the default and the most commonly used value, all the read are safely committed, and the read operations will never block each other. At some mission critical use cases (like using SlateDB as a Kafka-like at-least-once data bus), users may do not want to touch any unpersisted data which risks data loss, and they can set the watermark as `ReadWatermark::LastLocalPersisted` or `ReadWatermark::LastRemotePersisted` to achieve this goal.
-
-In the earlier draft, I've put a `ReadWatermark::LastCommitting` in the `ReadWatermark` enum. But after several discussions, we gradually reached a consensus that it's not a good idea to keep it, because it doesn't improve the performance of read operations, but only allows reading data that might be rolled back later, which might violate the semantics of conflict checking, and also introduce more corner cases to handle.
+```rust
+let opts = ReadOptions::new()
+    .with_durability_filter(DurabilityLevel::Local)
+    .with_dirty(true);
```

Comment:
nit: let's make the API an enum instead of boolean so we can evolve it if we ever have the need (e.g. `with_commit_level(CommitLevel::[DIRTY,CLEAN])`

Reply:
i hope the complexities will not increase on this aspect, at some points we have to limit the scope of the design space.

it's already uncommon for database engines to allow reading uncommitted data, and adding more options in this area will increase more complexity.

Reply:
I kinda prefer with_dirty as well. It seems intuitive to me: "dirty reads" are something I come across often when reading DB literature. "CommitLevel" makes me have to think more...

Reply:
ack, I'm onboard